### PR TITLE
Fix: .eslintignore paths should be absolute (fixes #5362)

### LINKED ISSUE #5362: .eslintignore paths should be absolute
Our .eslintignore patterns are recursive. This is one:
```
templates/**
```
is causing [issues](https://travis-ci.org/eslint/eslint/jobs/110553031#L503) with the build because it incorrectly matches `tests/templates/pr-create.md.ejs.js`

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: tests/templates/pr-create.md.ejs.js [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 "use strict";
 
+/* global describe, it */
```
Comment (alberto) [inline tests/templates/pr-create.md.ejs.js]: I'm not sure why this file needs this globals declaration while others don't (e.g. those in `test/lib/formatters`).

File: .eslintignore [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-conf/**
-coverage/**
-docs/**
-scripts/**
-templates/**
-tests/bench/**
-tests/fixtures/**
-tests/performance/**
-tmp/**
-test.js
```
Comment (alberto) [inline .eslintignore]: Please, note I removed
```
scripts/**
tests.js
```
since they don't exist in the codebase.

Comment (gyandeeps) [inline .eslintignore]: I would add `test.js` back as it is also ignored in `.gitignore`. The benefits of this file as its easy to do ad hoc eslint runs for different issue when you are working on eslint project and never worry about the file to get checked in or linted.

Comment (mysticatea) [inline .eslintignore]: I have `test.js` :)
It's included in `.gitignore` as well, so it's useful to verify issues.

Comment (alberto) [inline .eslintignore]: I didn't know that trick. :)

Let me know if `scripts` is used for something else I am not aware of.

Comment (gyandeeps) [inline .eslintignore]: Nope. You can remove scripts.

File: tests/templates/pr-create.md.ejs.js (hunk 1 above)
Comment (nzakas) [inline tests/templates/pr-create.md.ejs.js]: It probably doesn't. I might have accidentally copy-pasted from an older file.

Comment (nzakas) [inline tests/templates/pr-create.md.ejs.js]: Oh, I think those are added via an `.eslintrc` file in the `tests/lib` directory. We can probably move that file up to be under `tests/` instead, or else, copy it to `tests/templates`

Comment (alberto) [inline tests/templates/pr-create.md.ejs.js]: Thanks. I'll copy it to `tests/templates` since the rest of the folders don't need it.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**